File: test_ui.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  
  page.on('console', msg => console.log('BROWSER CONSOLE:', msg.type(), msg.text()));
  page.on('pageerror', error => console.log('BROWSER ERROR:', error.message));
  page.on('requestfailed', request => console.log('REQUEST FAILED:', request.url(), request.failure().errorText));
  page.on('response', async response => {
    if (!response.ok()) {
      console.log(`RESPONSE ERROR: ${response.status()} ${response.url()}`);
      try {
        console.log(await response.text());
      } catch (e) {}
    }
  });

  try {
    console.log("Navigating to Admin Panel movies page...");
    await page.goto('http://localhost:3000/movies', { waitUntil: 'networkidle' });
    console.log("Waiting a bit...");
    await page.waitForTimeout(2000);
    
    console.log("Navigating to Series page...");
    await page.goto('http://localhost:3000/series', { waitUntil: 'networkidle' });
    await page.waitForTimeout(2000);
    
    console.log("Navigating to Pages page...");
    await page.goto('http://localhost:3000/pages', { waitUntil: 'networkidle' });
    await page.waitForTimeout(2000);
    
    console.log("Test finished.");
  } catch (error) {
    console.error("TEST FAILED:", error);
  } finally {
    await browser.close();
  }
})();

File: test_ui.py
from playwright.sync_api import sync_playwright

def run():
    with sync_playwright() as p:
        browser = p.chromium.launch()
        page = browser.new_page()

        page.on("console", lambda msg: print(f"BROWSER CONSOLE: {msg.type} {msg.text}"))
        page.on("pageerror", lambda err: print(f"BROWSER ERROR: {err}"))
        
        def handle_response(response):
            if not response.ok:
                print(f"RESPONSE ERROR: {response.status} {response.url}")

        page.on("response", handle_response)

        try:
            print("Navigating to /movies...")
            page.goto("http://localhost:3000/movies", wait_until="networkidle")
            page.wait_for_timeout(2000)

            print("Navigating to /series...")
            page.goto("http://localhost:3000/series", wait_until="networkidle")
            page.wait_for_timeout(2000)

            print("Navigating to /pages...")
            page.goto("http://localhost:3000/pages", wait_until="networkidle")
            page.wait_for_timeout(2000)

            print("Test finished.")
        except Exception as e:
            print(f"TEST FAILED: {e}")
        finally:
            browser.close()

if __name__ == "__main__":
    run()
